messaging/repository/chat_repository.go

```diff
@@ -31,13 +35,57 @@ func (m *messageRepository) SaveMessage(ctx context.Context, message *domain.Mes
 }
 
 func (m *messageRepository) EditMessage(ctx context.Context, message *domain.Message) error {
-	panic("implement me")
+	var editedMsg domain.Message
+	_, err := m.dbSession.Query(editMessage, message.MessageBody, message.RoomID, message.SentTimestamp).WithContext(ctx).
+		ScanCAS(&editedMsg.RoomID, &editedMsg.SentTimestamp, &editedMsg.FromStudentID, &editedMsg.MessageBody)
+
+	if err != nil{
+		return err
+	}
+	return nil
 }
 
-func (m *messageRepository) GetMessages(ctx context.Context, roomID string, timeStamp time.Time) ([]domain.Message, error) {
-	panic("implement me")
+func (m *messageRepository) GetMessage(ctx context.Context, roomID string, timeStamp time.Time) (*domain.Message, error) {
+	var retrievedMsg domain.Message
+
+	err := m.dbSession.Query(getMessage, roomID, timeStamp).WithContext(ctx).
+		Scan(&retrievedMsg.RoomID, &retrievedMsg.SentTimestamp, &retrievedMsg.FromStudentID, &retrievedMsg.MessageBody)
+	if err != nil{
+		return nil, err
+	}
+
+	return &retrievedMsg, err
+}
+
+func (m *messageRepository) GetMessages(ctx context.Context, roomID string, timeStamp time.Time, limit int) ([]domain.Message, error) {
+	var retrievedMessages []domain.Message
+	var scanner gocql.Scanner
+
+	scanner = m.dbSession.Query(getMessages, roomID, timeStamp, limit).WithContext(ctx).Iter().Scanner()
+
+	for scanner.Next(){
+		var msg domain.Message
+		err := scanner.Scan(&msg.RoomID, &msg.SentTimestamp, &msg.FromStudentID, &msg.MessageBody)
+
+		if err == nil {
```

Comment:
log the error. Return the error as internal error, so they can make the call again. after err == nil {} block, append the message